lib/mobile_application_platform/security_token/service.rb

```diff
@@ -16,9 +16,19 @@ def token(application:, icn:)
         Rails.logger.info("#{config.logging_prefix} token success, application: #{application}, icn: #{icn}")
         parse_response(response, application, icn)
       rescue Common::Client::Errors::ClientError => e
+        description = case e
+                      in _msg, 500, _body
+                        'service returned Internal Server error most likely due to an invalid ICN'
+                      in _msg, (400|401), { error: }
+                        error
+                      in _msg, (400|401), { error_description: }
+                        error_description
+                      else
+                        'unknown'
+                      end
```

Comment:
I think would be cleaner and more maintainable to create new error classes in [here](https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/mobile_application_platform/security_token/errors.rb) for these different scenarios and handle the logic of which error to raise in a base error class. Here is an example of what I'm talking about - [error.rb](https://github.com/department-of-veterans-affairs/sign-in-service-rb/blob/main/lib/sign_in_service/error.rb) (won't be exactly like this)

Reply:
Nice, I agree. Plus, you get the added benefit of raising an error more descriptive than the generic client error.